Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python312/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python312/lib/python3.12/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python312/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python312/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python312/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python312/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python312/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
pythonstudierende = [
    Student("Abendrot", "Anna", "Maschinenbau", 3, "12345678"),
    Student("Berliner", "Bob", "Maschinenbau", 5, "87654321"),
    Student("Colorado", "Charlie", "Elektrotechnik", 2, "11223344")
]

# alle Namen ausgeben
for student in studierende:
    print(f"{student.vorname} {student.nachname}")
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[8], line 8[0m
[1;32m      1[0m pythonstudierende [38;5;241m=[39m [
[1;32m      2[0m     Student([38;5;124m"[39m[38;5;124mAbendrot[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mAnna[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mMaschinenbau[39m[38;5;124m"[39m, [38;5;241m3[39m, [38;5;124m"[39m[38;5;124m12345678[39m[38;5;124m"[39m),
[1;32m      3[0m     Student([38;5;124m"[39m[38;5;124mBerliner[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mBob[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mMaschinenbau[39m[38;5;124m"[39m, [38;5;241m5[39m, [38;5;124m"[39m[38;5;124m87654321[39m[38;5;124m"[39m),
[1;32m      4[0m     Student([38;5;124m"[39m[38;5;124mColorado[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mCharlie[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mElektrotechnik[39m[38;5;124m"[39m, [38;5;241m2[39m, [38;5;124m"[39m[38;5;124m11223344[39m[38;5;124m"[39m)
[1;32m      5[0m ]
[1;32m      7[0m [38;5;66;03m# alle Namen ausgeben[39;00m
[0;32m----> 8[0m [38;5;28;01mfor[39;00m student [38;5;129;01min[39;00m [43mstudierende[49m:
[1;32m      9[0m     [38;5;28mprint[39m([38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00mstudent[38;5;241m.[39mvorname[38;5;132;01m}[39;00m[38;5;124m [39m[38;5;132;01m{[39;00mstudent[38;5;241m.[39mnachname[38;5;132;01m}[39;00m[38;5;124m"[39m)

[0;31mNameError[0m: name 'studierende' is not defined

